src/libraries/System.Net.Quic/ref/System.Net.Quic.cs

```diff
@@ -32,16 +32,28 @@ public sealed partial class QuicConnection : System.IDisposable
         public System.Threading.Tasks.ValueTask<System.Net.Quic.QuicStream> OpenBidirectionalStreamAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public System.Threading.Tasks.ValueTask<System.Net.Quic.QuicStream> OpenUnidirectionalStreamAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
-    public partial class QuicConnectionAbortedException : System.Net.Quic.QuicException
+    public enum QuicError
     {
-        public QuicConnectionAbortedException(string message, long errorCode) : base(default(string)) { }
-        public long ErrorCode { get { throw null; } }
+        Success = 0,
+        InternalError = 1,
+        ConnectionAborted = 2,
+        StreamAborted = 3,
+        AddressInUse = 4,
+        InvalidAddress = 5,
+        ConnectionTimeout = 6,
+        HostUnreachable = 7,
+        ConnectionRefused = 8,
+        VersionNegotiationError = 9,
+        ConnectionIdle = 10,
+        ProtocolError = 11,
+        OperationAborted = 12,
     }
-    public partial class QuicException : System.Exception
+    public sealed partial class QuicException : System.IO.IOException
     {
-        public QuicException(string? message) { }
-        public QuicException(string? message, System.Exception? innerException) { }
-        public QuicException(string? message, System.Exception? innerException, int result) { }
+        public QuicException(System.Net.Quic.QuicError error, string message, long? applicationErrorCode, System.Exception? innerException) { }
+        public QuicException(System.Net.Quic.QuicError error, string message, long? applicationErrorCode, System.Exception? innerException, int result) { }
```

Comment:
This ctor is not part of the approved API. Either we need to remedy that or just make it internal.

Reply:
Removed, I forgot that `HResult` setter is public